pkg/controller/nmstate/nmstate_controller_test.go

```diff
@@ -69,4 +129,36 @@ var _ = Describe("NMState controller reconcile", func() {
 			Expect(result).To(Equal(reconcile.Result{}))
 		})
 	})
+	Context("node selector is set in Operator spec", func() {
+		var (
+			request reconcile.Request
+		)
+		BeforeEach(func() {
+			s := scheme.Scheme
+			s.AddKnownTypes(nmstatev1alpha1.SchemeGroupVersion,
+				&nmstatev1alpha1.NMState{},
+			)
+			// set NodeSelector field in operator Spec
+			nmstate.Spec.NodeSelector = dsNodeSelector
+			objs := []runtime.Object{&nmstate}
+			// Create a fake client to mock API calls.
+			cl = fake.NewFakeClientWithScheme(s, objs...)
+			reconciler.client = cl
+			request.Name = existingNMStateName
+		})
+		It("NodeSelector labels should be defined for Daemonsets", func() {
+			result, err := reconciler.Reconcile(request)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(result).To(Equal(reconcile.Result{}))
+			// verify NodeSelector value for both worker and master DS
+			ds := &appsv1.DaemonSet{}
```

Comment:
You can do cl.List and filter by namespace I think

Reply:
Yes, it's more elegant to List DS here, but I'm getting [1] error when trying to List DS [2]
Any idea what am I missing here?

[1] tems[0] can't assign or convert unstructured.Unstructured into v1.DaemonSet
[2]
                        dsList := appsv1.DaemonSetList{}
                        err := cl.List(context.TODO(), &dsList, &client.ListOptions{ Namespace: "nmstate"})

Reply:
Can we test with client.InNamespace [1] ?

[1] https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/client_test.go#L1793